src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonConverter.Sync.cs

```diff
@@ -3,88 +3,181 @@
             int arrayIndex = 0;
 
-            ReadData(ref reader, options, returnType, ref current, ref previous, ref arrayIndex);
+            FromJsonObjectState current = default;
+            JsonClassInfo classInfo = settings.GetOrAddClass(returnType);
+            current.ClassInfo = classInfo;
+            if (classInfo.ClassType != ClassType.Object)
+            {
+                current.PropertyInfo = classInfo.GetPolicyProperty();
+            }
+
+            FromJson(ref reader, settings, returnType, ref current, ref previous, ref arrayIndex);
 
             return current.ReturnValue;
         }
+
+        public static string ToJsonString(object value, JsonConverterSettings settings = null)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+
+            Span<byte> jsonBytes = ToJsonInternal(value, settings);
+#if BUILDING_INBOX_LIBRARY
+            string stringJson = s_utf8Encoding.GetString(jsonBytes);
+#else
+            string stringJson;
+            if (jsonBytes.IsEmpty)
+            {
+                stringJson = string.Empty;
+            }
+            unsafe
+            {
+                fixed (byte* bytePtr = jsonBytes)
+                {
+                    stringJson = s_utf8Encoding.GetString(bytePtr, jsonBytes.Length);
+                }
+            }
+#endif
+            return stringJson;
+        }
+
+        public static Span<byte> ToJson(object value, JsonConverterSettings settings = null)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+
+            return ToJsonInternal(value, settings);
+        }
+
+        private static Span<byte> ToJsonInternal(object value, JsonConverterSettings settings = null)
+        {
+            if (settings == null)
+                settings = s_DefaultSettings;
+
+            ToJsonObjectState current = default;
+            Type initialType = value.GetType();
+
+            JsonClassInfo classInfo = settings.GetOrAddClass(initialType);
+            current.ClassInfo = classInfo;
+            current.CurrentValue = value;
+            if (classInfo.ClassType != ClassType.Object)
+            {
+                current.PropertyInfo = classInfo.GetPolicyProperty();
+            }
+
+            List<ToJsonObjectState> previous = null;
+            int arrayIndex = 0;
+
+            var state = new JsonWriterState(settings.WriterOptions);
+
+            byte[] result;
+
+            using (var output = new ArrayBufferWriter(settings.DefaultBufferSize))
+            {
+                var writer = new Utf8JsonWriter(output, state);
+
+                ToJson(ref writer, settings, ref current, ref previous, ref arrayIndex);
+
+                int byteCount;
+                checked
+                {
+                    byteCount = (int)writer.BytesWritten;
+                }
+
+                Span<byte> json = output.GetSpan().Slice(0, byteCount);
```

Comment:
`GetSpan` gives you the span to write more data into, not what has already been written.

What you should use is `output.Formatted` instead.

Reply:
Is this called `WrittenMemory` in the newer version of ArrayBufferWriter?

Reply:
Yes.